File: Domain/Event.cs
namespace Domain
{
    public class Event
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; } 
        public string Description { get; set; }
        public string Location { get; set; }
        public string Image { get; set; }
        public bool IsCancelled { get; set; }
        public ICollection<EventAttendee> Attendees { get; set; } = new List<EventAttendee>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: client-app/src/app/models/event.ts
import { Profile } from "./profile"

export interface Event {
    id: string
    title: string
    date: Date | null
    description: string
    location: string
    image: string
    hostUsername: string
    isCancelled: boolean
    isGoing: boolean
    isHost: boolean
    host?: Profile
    attendees: Profile[]
}

export class Event implements Event {
    constructor(init?: EventFormValues) {
        Object.assign(this, init);
    }
}

export class EventFormValues {
    id?: string = undefined;
    title: string = '';
    description: string = '';    
    date: Date | null = null;
    location: string = '';
    image: string = '';

    constructor(event?: EventFormValues) {
        if (event) {
            this.id = event.id;
            this.title = event.title;
            this.description = event.description;
            this.date = event.date;
            this.location = event.location;
            this.image = event.image;
        } 
    }
}

export interface FilterEvent {
    id?: string;
    title: string;
    location: string;
}